extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -272,4 +274,27 @@ void start(
             recorder.start(shutdown, main.getInstance());
         }
     }
+
+    /**
+     * Build steps related to Camel Attachments.
+     */
+    public static class Attachments {
+
+        /**
+         * Produces an {@link UploadAttacherBuildItem} holding a no-op {@link UploadAttacher}.
+         * <p>
+         * Note that this {@link BuildStep} is effective only if {@code camel-quarkus-attachments} extension is not in
+         * the class path.
+         *
+         * @param recorder the {@link CoreAttachmentsRecorder}
+         * @return a new {@link UploadAttacherBuildItem}
+         */
+        @Overridable
+        @Record(value = ExecutionTime.STATIC_INIT, optional = true)
+        @BuildStep(onlyIfNot = Flags.AttachmentsAvailable.class)
```

Comment:
Don't think this flag is needed as the build item is marked as @Overridable, thus is is overridden by the build item from the attachment extension

Reply:
I tried without that and it complained that it was produced twice. Let me try again.

Reply:
You need to mark the recorder optional = true on both build items afair

Reply:
Hm... It works without the flag, but `optional = true` must be there. Otherwise it throws `Cannot provide/consume multiple values for class org.apache.camel.quarkus.core.deployment.UploadAttacherBuildItem` That's a bit strange. I would not spontaneously consider UploadAttacherBuildItem to be optional. One or the other provider is always there.

Anyway, thanks for the catch and let me remove the flag.

Reply:
Yes, there's an issue I think for that